src/data-workspace/validation/validation-rule-expression.js

```diff
@@ -20,9 +10,45 @@ const ValidationRuleExpression = ({ validationRule }) => {
         metaData: { leftSide, rightSide, operator } = {},
     } = validationRule
 
-    return `${leftSide.displayDescription} (${leftsideValue}) ${
-        operatorDisplay[operator] ?? operator
-    } ${rightSide.displayDescription} (${rightsideValue})`
+    const { displayDescription: leftDescription } = leftSide
+    const { displayDescription: rightDescription } = rightSide
+
+    const operatorDisplay = {
+        equal_to: '==',
+        not_equal_to: '!=',
+        greater_than: '>',
+        greater_than_or_equal_to: '>=',
+        less_than: '<',
+        less_than_or_equal_to: '<=',
+        compulsory_pair: i18n.t(
+            '{{leftDescription}} cannot be empty when {{rightDescription}} has a value',
+            { leftDescription, rightDescription }
+        ),
+        exclusive_pair: i18n.t(
+            '{{leftDescription}} cannot have a value when {{rightDescription}} has a value',
+            { leftDescription, rightDescription }
+        ),
+    }
+
+    if (operator === 'compulsory_pair' || operator === 'exclusive_pair') {
+        return (
+            <span className={styles.formulaValue}>
+                {operatorDisplay[operator]}
+            </span>
+        )
+    }
```

Comment:
Just a suggestion: There are already conditions checking whether it's one of these "special" operators, so we could keep the static ones outside of this component to keep the component itself smaller / more readable. I wouldn't mind having a component for the "pair" operators, not sure what a good name would be though (`<ConditionalRelationshipOperator />` sounds awkward and is quite long...).

Reply:
updated .. not sure about the naming either, but I think regardless of the operator, one of them is an expression with right and left-side values, while the other one is a kind of static description. Have [a look](https://github.com/dhis2/data-entry-app/compare/42cbfb42774908ec0c4c1d5ad338ca962f1c07d0..cf9798cacf002dfdcaf29040ef643a40f6dce9a0) and let me know what you think.